src/include/ipc/probe.h

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+/**
+ * \brief Definitions of shifts and masks for format encoding in probe
+ *	  extraction stream
+ *
+ * Audio format from extraction probes is encoded as 32 bit value. Following
+ * graphic explains encoding.
+ *
+ * A|BBBB|CCCC|DDDD|EEEEE|FF|GG|H|I|J|XXXXXXX
+ * A - 1 bit - Specifies Type Encoding - 1 for Standard encoding
+ * B - 4 bits - Specify Standard Type - 0 for Audio, 1 for Video
+ * C - 4 bits - Specify Audio format - 0 for PCM, 1 for MP3
+ * D - 4 bits - Specify Sample Rate - value enumerating standard sample rates:
+ *				      8000 Hz		= 0x0
+ *				      11025 Hz		= 0x1
+ *				      12000 Hz		= 0x2
+ *				      16000 Hz		= 0x3
+ *				      22050 Hz		= 0x4
+ *				      24000 Hz		= 0x5
+ *				      32000 Hz		= 0x6
+ *				      44100 Hz		= 0x7
+ *				      48000 Hz		= 0x8
+ *				      64000 Hz		= 0x9
+ *				      88200 Hz		= 0xA
+ *				      96000 Hz		= 0xB
+ *				      128000 Hz		= 0xC
+ *				      176400 Hz		= 0xD
+ *				      192000 Hz		= 0xE
+ *				      none of the above = 0xF
+ * E - 5 bits - Specify Number of Channels minus 1
+ * F - 2 bits - Specify Sample Size, number of valid sample bytes
+ * G - 2 bits - Specify Container Size, number of container bytes
```

Comment:
how does it work with 32-bit containers? there must be a missing encoding here.

Reply:
We should really also be reusing the same type/structs for hw_param format configuration here too. That we we dont need to define two format APIs.

Reply:
@lgirdwood That's the interface that we got from cAVS and it was documented in the FDK, as far as I know. We were tasked to match it, so we didn't take any actions to change that.

Reply:
documentation updated